### ISSUE TITLE
Remove `axios.all` and `axios.spread`

### ISSUE THREAD

**Issue by rubennorte** [axios/axios#1042]
Those helpers can easily be replaced with native features (`Promise.all` and ES6 parameter destructuring) or specialized libraries, so we should restrict our API to HTTP-related methods.

We can deprecate them in 0.x and completely remove then in 1.0.

**Comment by louisukiri** [comment]
Promise.all is not supported in IE at all. Will we then need to use a polyfill?

**Comment by rubennorte** [comment]
Axios uses `Promise.all` under the hood so you must be polyfilling it already.

**Comment by qm3ster** [comment]
Should the following in the README example:
```js
axios.all([getUserAccount(), getUserPermissions()])
  .then(axios.spread(function (acct, perms) {
    // Both requests are now complete
  }));
```
be replaced with this instead:
```js
Promise.all([getUserAccount(), getUserPermissions()])
  .then(function ([acct, perms]) {
    // Both requests are now complete
  });
```
or this:
```js
const [acct, perms] = await Promise.all([getUserAccount(), getUserPermissions()])
// Both requests are now complete
```

**Comment by ChadTaljaardt** [comment]
@qm3ster i think the second version is better. 
```
const [acct, perms] = await Promise.all([getUserAccount(), getUserPermissions()])
```

**Comment by qm3ster** [comment]
@ChadTaljaardt there's `await` in the docs already, but the `.all` example is still `axios.all` :pizza: 

https://github.com/axios/axios/blob/master/README.md#example

**Comment by ohtaeg** [comment]
I'm interested in #2260 . Can I try?

**Comment by jasonsaayman** [comment]
Resolved by #2727

**Comment by Jossnaz** [comment]
if someone is looking for a typescript solution:

```
                const [cat_res, man_res, p_res, cm_res] = await Promise.all<{ data: Category[] },  { data: Manufacturer[] }, { data: Product[] }, { data: Colormatrix }>([
                    Axios.get('/admin/subcategories/read'),
                    Axios.get('/admin/manufacturers/read'),
                    Axios.get('/admin/products/read'),
                    Axios.get('/admin/colormatrices/read'),
                ]);

```
